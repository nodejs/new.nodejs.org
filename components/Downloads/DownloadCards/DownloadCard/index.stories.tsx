import DownloadCard from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof DownloadCard>;
type Meta = MetaObj<typeof DownloadCard>;

export const Selected: Story = {
  args: {
    os: 'MAC',
    icon: 'mac-download-logo.svg',
    label: 'MAC Installer',
    download: 'https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg',
    filename: 'node-v18.15.0.pkg',
    selected: true,
    onSelect: () => {},
  },
};

export const NotSelected: Story = {
  args: {
    os: 'MAC',
    icon: 'mac-download-logo.svg',
    label: 'MAC Installer',
    download: 'https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg',
    filename: 'node-v18.15.0.pkg',
    selected: false,
    onSelect: () => {},
  },
};

export default { component: DownloadCard } as Meta;
