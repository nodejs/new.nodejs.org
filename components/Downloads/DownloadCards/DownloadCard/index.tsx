import classNames from 'classnames';
import { MdGetApp } from 'react-icons/md';
import { Tab } from 'react-tabs';
import Image from 'next/image';
import styles from './index.module.scss';
import { useRouter } from '@/hooks/useRouter';
import type { FC } from 'react';
import type { UserOS } from '@/types/userOS';

type DownloadCardProps = {
  os: UserOS;
  icon: string;
  label: string;
  download: string;
  filename: string;
  selected: boolean;
  onSelect: (os: string) => void;
};

const DownloadCard: FC<DownloadCardProps> = ({
  os,
  icon,
  label,
  download,
  filename,
  selected,
  onSelect,
}) => {
  const { basePath } = useRouter();

  const tabClassNames = classNames(styles.downloadCard, {
    [styles.downloadCardActive]: selected,
  });
  const selectCard = () => onSelect(os);
  const size = selected ? 56 : 48;

  return (
    <Tab
      className={tabClassNames}
      id={`download-card-${os}`}
      key={os}
      onClick={selectCard}
      selectedClassName={styles.downloadCardActive}
      tabIndex="0"
    >
      <div className={styles.top}>
        <Image
          height={size}
          width={size}
          src={`${basePath}/static/images/logos/${icon}`}
          alt={label}
        />
        {selected && (
          <a className={styles.link} href={download}>
            <MdGetApp />
          </a>
        )}
      </div>
      <p className={styles.label}>{label}</p>
      <p className={styles.filename} data-testid="lts">
        {filename}
      </p>
    </Tab>
  );
};

export default DownloadCard;
