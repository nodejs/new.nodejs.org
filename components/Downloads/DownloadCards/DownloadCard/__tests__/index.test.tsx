import { render } from '@testing-library/react';
import DownloadCard from './../index';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    basePath: '',
  })),
}));

describe('DownloadCard component', () => {
  it('selected renders correctly', () => {
    const { container } = render(
      <DownloadCard
        name="MAC"
        icon="mac-download-logo.svg"
        label="MAC Installer"
        download="https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg"
        filename="node-v18.15.0.pkg"
        selected={true}
        onSelect={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('not selected renders correctly', () => {
    const { container } = render(
      <DownloadCard
        name="MAC"
        icon="mac-download-logo.svg"
        label="MAC Installer"
        download="https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg"
        filename="node-v18.15.0.pkg"
        selected={false}
        onSelect={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
