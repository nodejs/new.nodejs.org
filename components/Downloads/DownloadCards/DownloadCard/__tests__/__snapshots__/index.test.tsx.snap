// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadCard component not selected renders correctly 1`] = `
<div>
  <li
    aria-controls="paneldownload-card-MAC"
    aria-disabled="false"
    aria-selected="false"
    class="downloadCard"
    data-rttab="true"
    id="tabdownload-card-MAC"
    role="tab"
    tabindex="0"
  >
    <div
      class="top"
    >
      <img
        alt="MAC Installer"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/static/images/logos/mac-download-logo.svg"
        style="color: transparent;"
        width="48"
      />
    </div>
    <p
      class="label"
    >
      MAC Installer
    </p>
    <p
      class="filename"
      data-testid="lts"
    >
      node-v18.15.0.pkg
    </p>
  </li>
</div>
`;

exports[`DownloadCard component selected renders correctly 1`] = `
<div>
  <li
    aria-controls="paneldownload-card-MAC"
    aria-disabled="false"
    aria-selected="false"
    class="downloadCard downloadCardActive"
    data-rttab="true"
    id="tabdownload-card-MAC"
    role="tab"
    tabindex="0"
  >
    <div
      class="top"
    >
      <img
        alt="MAC Installer"
        data-nimg="1"
        decoding="async"
        height="56"
        loading="lazy"
        src="/static/images/logos/mac-download-logo.svg"
        style="color: transparent;"
        width="56"
      />
      <a
        class="link"
        href="https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
          />
        </svg>
      </a>
    </div>
    <p
      class="label"
    >
      MAC Installer
    </p>
    <p
      class="filename"
      data-testid="lts"
    >
      node-v18.15.0.pkg
    </p>
  </li>
</div>
`;
